sale_order_line_date/tests/test_sale_order_line_date.py

```diff
@@ -0,0 +1,161 @@
+# © 2016 OdooMRP team
+# © 2016 AvanzOSC
+# © 2016 Serv. Tecnol. Avanzados - Pedro M. Baeza
+# © 2016-22 ForgeFlow S.L. (https://forgeflow.com)
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import datetime
+
+from odoo import fields
+from odoo.tests.common import SavepointCase
+
+
+class TestSaleOrderLineDates(SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        """Setup a Sale Order with 4 lines."""
+        super().setUpClass()
+        customer = cls.env.ref("base.res_partner_3")
+        cls.company = cls.env.ref("base.main_company")
+        cls.company.security_lead = 1
+
+        price = 100.0
+        qty = 5
+        product_id = cls.env.ref("product.product_product_7")
+        cls.today = fields.Datetime.now()
+        cls.dt1 = cls.today + datetime.timedelta(days=9)
+        cls.dt2 = cls.today + datetime.timedelta(days=10)
+        cls.dt3 = cls.today + datetime.timedelta(days=3)
+        cls.sale1 = cls._create_sale_order(customer, cls.dt2)
+        cls.sale_line1 = cls._create_sale_order_line(
+            cls.sale1, product_id, qty, price, cls.dt1
+        )
+        cls.sale_line2 = cls._create_sale_order_line(
+            cls.sale1, product_id, qty, price, cls.dt2
+        )
+        cls.sale_line3 = cls._create_sale_order_line(
+            cls.sale1, product_id, qty, price, None
+        )
+        cls.sale2 = cls._create_sale_order(customer, cls.dt2)
+        cls.sale_line4 = cls._create_sale_order_line(
+            cls.sale2, product_id, qty, price, cls.dt3
+        )
+        cls.sale_line5 = cls._create_sale_order_line(
+            cls.sale2, product_id, qty, price, cls.dt2
+        )
+        cls.sale_line6 = cls._create_sale_order_line(
+            cls.sale2, product_id, qty, price, cls.dt1
+        )
+
+    def _create_sale_order(self, customer, date):
```

Comment:
The tests are failing because you are using now this function inside setUpCls. Then, please add this function the @classmethod decorator and change self by cls. Same with the _create_sale_order_line method
![image](https://user-images.githubusercontent.com/76053579/233025442-9e814d62-71f2-4f5f-b7f9-3ea023334f55.png)

Reply:
@BT-anieto thank you, I fixed it